Sources/SwiftDocC/Infrastructure/Topic Graph/AutomaticCuration.swift

```diff
@@ -171,17 +172,30 @@ public struct AutomaticCuration {
             return nil
         }
         
+        let variantLanguageIDs = Set(variantsTraits.compactMap(\.interfaceLanguage))
+        
+        func isRelevant(_ filteredGroup: DocumentationContentRenderer.ReferenceGroup) -> Bool {
+            // Check if the task group is filtered to a subset of languages
+            if let languageFilter = filteredGroup.languageFilter,
+               !languageFilter.contains(where: { variantLanguageIDs.contains($0.id) })
+            {
+                // This group is only applicable to other languages than the given variant traits.
+                return false
+            }
+            
+            // Otherwise, check that the group contains the this reference.
+            return filteredGroup.references.contains(node.reference)
+        }
+        
         func filterReferences(_ references: [ResolvedTopicReference]) -> [ResolvedTopicReference] {
             Array(
                 references
-                // Don't include the current node.
-                .filter { $0 != node.reference }
-            
-                // Filter out nodes that aren't available in any of the given traits.
                 .filter { reference in
-                    context.sourceLanguages(for: reference).contains(where: { language in
-                        variantsTraits.contains(where: { $0.interfaceLanguage == language.id})
-                    })
+                    // Don't include the current node.
+                    reference != node.reference
+                    &&
+                    // Filter out nodes that aren't available in any of the given traits.
+                    context.sourceLanguages(for: reference).contains(where: { variantLanguageIDs.contains($0.id) })
```

Comment:
Is this just refactoring what we had earlier? Would it be even simpler to intersect the arrays somehow, vs. a nested loop? Not really a concern since the number of languages is usually 1 or 2. Would just be more readable.

Maybe something like this: 


```
let variantLanguages = Set(variantsTraits.compactMap{ $0.interfaceLanguage }.map{ SourceLanguage(id: $0)})

etc...


// Filter out nodes that aren't available in any of the given traits.
!context.sourceLanguages(for: reference).intersection(variantLanguages).isEmpty

```


Although you use `variantLanguageIDs` above in `isRelevant` so we'd need some more refactoring.

Reply:
Yes, I noticed that the two separate `filters` could be expressed as once combined condition and since `isRelevant` also does a language check I wanted both places to do the same check (and felt that it was nice to get change one of the `contains(where:)` to a simpler `contains(_:)`.

Because the languages is going to be incredibly small (either 1 or 2) any performance difference is not going to be noticeable so I can update it to use `Set<SourceLanguage>` in both places for readability.

Reply:
I updated to use a set of languages in https://github.com/swiftlang/swift-docc/pull/1082/commits/693910913d772eeecff239022f554ac48130d720.